### PR TITLE
Create a lease for the snapshots we prepare

### PR DESCRIPTION
**Pull request by rumpl**
With a lease we make sure that containerd won't garbage collect out
snapshot. We remove the lease when the container is removed and leave it
to containerd to clean things up.

### CODE REVIEW

**Review by vvoland — APPROVED** [review]

File: daemon/create.go [hunk 1]
```diff
@@ -193,6 +194,16 @@ func (daemon *Daemon) create(ctx context.Context, opts createOpts) (retC *contai
...
+		// Add a lease so that containerd doesn't garbage collect our snapshot
+		ls := daemon.containerdCli.LeasesService()
+		lease, err := ls.Create(ctx, leases.WithID(ctr.ID))
+		if err != nil {
+			return nil, err
+		}
+		ls.AddResource(ctx, lease, leases.Resource{
+			ID:   ctr.ID,
+			Type: "snapshots/" + daemon.graphDriver,
+		})
```
Comment (thaJeztah) [inline daemon/create.go:206]: Discussing with @vvoland - possibly with this fix in place, we can revert the changes from https://github.com/rumpl/moby/pull/47, which will make upstreaming our changes easier
